frappe/erpnext
# feat: depreciate Asset after sale

## Proposed Changes:

If an Asset is set to depreciate on December 31 of every year, but is sold on June 30, it should undergo another depreciation, for the six months that passed since the last depreciation. This feature:

- Ensures that this depreciation is carried out.
- Reverses it and resets the Depreciation Schedule on returning the Asset.

![Screenshot 2021-07-19 at 9 19 46 AM](https://user-images.githubusercontent.com/25903035/126100152-aed1030d-5bf7-4c3b-ae76-d675cb222d71.png)

![Screenshot 2021-07-19 at 10 58 45 AM](https://user-images.githubusercontent.com/25903035/126107548-074d2151-eac4-4956-a42d-cfb79c29e8bd.png)

<details>
<summary>Testing Info</summary>

1. Create a Fixed Asset Item(Check the _Is Fixed Asset_ box in Item).
2. Create an Asset linked with the above Item.
3. Check the _Calculate Depreciation_ box and enter _Finance Books_ details for the Asset.
4. Save and Submit the Asset.
[A _Depreciation Schedule_ will be displayed at this point]
5. Create a Sales Invoice for this Asset.
6. After submitting the Invoice, check the _Depreciation Schedule_ for the Asset.
7. Submit a Credit Note against the Sales Invoice that was just created.
8. Check the _Depreciation Schedule_ again. It should have gone back to its original state.

</details>

`no-docs`
